webui-ros-joystick/nodejs/main.js

```diff
@@ -137,31 +137,32 @@ rosnodejs
   .initNode("/rosnodejs")
   .then(async (rosNode) => {
 
-    cmdVelPublisher = rosNode.advertise(
-      "/cmd_vel",
-      geometryMsgs.Twist,
-      defaultPublisherOptions
-    );
-
     let privateNH = rosnodejs.getNodeHandle(rosNode.getNodeName());
+    let waitNodesStr = await getRosParam(privateNH, "wait_nodes", "");
+    let cmdVelTopic = await getRosParam(privateNH, "cmd_vel_topic", "/cmd_vel");
     maxLinVel = await getRosParam(privateNH, "max_lin_vel", 1.0);
     maxAngVel = await getRosParam(privateNH, "max_ang_vel", 1.0);
     maxLinAccel = await getRosParam(privateNH, "max_lin_accel", 2.0);
     maxAngAccel = await getRosParam(privateNH, "max_ang_accel", 2.0);
     eStopPresent = await getRosParam(privateNH, "e_stop", false);
-    let waitNodesStr = await getRosParam(privateNH, "wait_nodes", "");
+
+    cmdVelPublisher = rosNode.advertise(
+      cmdVelTopic,
+      geometryMsgs.Twist,
+      defaultPublisherOptions
+    );
 
     if (eStopPresent) {
       eStop = true;
 
       eStopSubscriber = rosNode.subscribe(
-        "/e_stop",
+        "e_stop",
```

Comment:
The e_stop changes are not related to the task, are they intentional?

Reply:
I know, but as I was looking at topics naming i changed that so namespaces could be used. So I added this quick fix :P

Reply:
Same as above